modules/apps/transfer/internal/convert/convert.go

```diff
@@ -0,0 +1,54 @@
+package convert
+
+import (
+	"strings"
+
+	"github.com/cosmos/ibc-go/v8/modules/apps/transfer/types"
+)
+
+// PacketDataV1ToV2 converts a v1 packet data to a v2 packet data.
+func PacketDataV1ToV2(packetData types.FungibleTokenPacketData) types.FungibleTokenPacketDataV2 {
+	if err := packetData.ValidateBasic(); err != nil {
+		panic(err)
+	}
+
+	v2Denom, trace := ExtractDenomAndTraceFromV1Denom(packetData.Denom)
+	return types.FungibleTokenPacketDataV2{
+		Tokens: []types.Token{
+			{
+				Denom:  v2Denom,
+				Amount: packetData.Amount,
+				Trace:  trace,
+			},
+		},
+		Sender:   packetData.Sender,
+		Receiver: packetData.Receiver,
+		Memo:     packetData.Memo,
+	}
+}
+
+// extractDenomAndTraceFromV1Denom extracts the base denom and remaining trace from a v1 IBC denom.
+func ExtractDenomAndTraceFromV1Denom(v1Denom string) (string, []string) {
```

Comment:
this function appears to assume the denom is validated, should we add this in the godoc?

Otherwise the parsing logic doesn't account for edge cases like the other [function](https://github.com/cosmos/ibc-go/blob/feat/ics20-v2/modules/apps/transfer/types/trace.go#L104) does

Reply:
I think it's fine for now, lets revisit after #6362 and #6221 which I think will cleanup all the parsing logic. After that issue we should only parse from full path string to typed trace and then have a function to glue them together again